R/rj.b.R

```diff
@@ -165,13 +172,39 @@ RjClass <- R6::R6Class(
 
             } else if (os == 'Windows') {
 
-                entries <- try(readRegistry('Software\\R-core\\R', 'HLM'))
-                if (inherits(entries, 'try-error'))
+                regHLM <- file.path("SOFTWARE", "R-core", "R64", fsep = "\\")
+                entries <- try(readRegistry(regHLM, 
+                                            hive = "HLM",
+                                            maxdepth = 2, 
+                                            view = "64-bit"))
+
+                if ( inherits(entries, 'try-error') )
                     stop('Could not find system R (no appropriate registry entries)')
-                home <- entries$InstallPath
-                path <- file.path(home, 'bin', 'x64', 'R.exe')
-                if (file.exists(path))
-                    return(path)
+
+                if ( is.null(entries$`Current Version`) || is.null(entries$InstallPath) ) {
+
+                    ssDx <- function(x, n) {sapply(x, function(xx) substr(xx, (nchar(xx)-n+1), nchar(xx)) )}
+
+                    path <- file.path(entries[[1]]$InstallPath)
+                    rcv <- ssDx(path, 5)
```

Comment:
provide a short description of what you're doing here with the `ssDx()` function ... i have no idea :P

Reply:
It is a simple `SubStringRight()` to retrieve the last 5 characters of the path which will serve version of R as major.minor.pathc to use for the library :P